### DISCUSSION TITLE
Is Supabase suitable for IoT / Time-series

### DISCUSSION THREAD

**Discussion by fullmetal-fred** [supabase/supabase#13065] [Questions]
I'm considering using Supabase as the complete backend for my app. This app has an IoT component, meaning thousands of incoming data points every minute, and thousands of devices subscribed for real-time updates. Traditionally I'd use a standard Pubsub (like MQTT etc) for send/receiving data, and a timeseries database (like InfluxDB) for ingesting all of the metrics. Buuuuut Supabase is really cool...so I'd love to use it if performance is going to allow!

**Comment by organicnz** [comment]
I'd love to use it on Nvidia Jetson Nano or Raspberry Pi :)

**Comment by haykodarb** [comment]
Researching this topic I found out you can enable the timescale-db extension for properly scaling the Postgres database for IoT applications.
 
https://supabase.com/docs/guides/database/extensions/timescaledb

My guess would be the current best way to develop IoT with Supabase would be:

- Create a Supabase database with timescaledb enabled.
- Open a MQTT broker.
- Subscribe to that MQTT broker on a NodeJS (or whatever) server. Write code to read topic changes and post to Supabase.

You can check performance data on timescaledb's web page.

https://www.timescale.com/blog/timescaledb-vs-6a696248104e/

**Comment by fullmetal-fred** [comment]
Thanks for the response here! I’ll reach back out with how it goes!
________________________________

**Reply by RodolpheBeloncle** [reply]
hi @avfuturist , i would like to know your feedback after your research. I've got a project already running on react app and as well ionic mobile app. I've got as well a mosquitto broker which is running locally.All 's working find , i wanted to use firebase at the beginning  but i rather like open source tools.My goal is to connect the broker to a stream database like supabase.Any recommendation or help would be very nice.Have a good day

**Reply by fullmetal-fred** [reply]
@RodolpheBeloncle I can't confirm if it's fully appropriate for an IoT use-case yet mostly because I haven't been able to really load test anything. I'll say from a development perspective Supabase is the best. I love it and it's changed the way I build backends. I will say that between the realtime engine and Postgres TimescaleDB tables, I see zero reason why Supabase won't work...there's plenty of precedent on Postgres. For the time being we haven't been using MQTT yet...just middleware and the JS client, but I have ambitions to run edge functions for MQTT subscribers & publishers. Sorry I'm not more help yet. I will say this: I wouldn't fathom trying anything else at this point. (username was changed from avfuturist, fyi)

**Comment by ShanArosh** [comment]
supabase provides timescaledb as an extension for handling time-series data.

The problem happens when you need to use timescale Continuous aggregates, Distributed hypertables, Compression, Data retention,Actions and automation. Supabase can only provide TimescaleDB Apache 2 Edition due to timescale licensing.

You'll need to edit postgresql.conf file or alter system commands to change the license of timescale to use the above features.
So self-hosting is the only option I found. If anyone found a way around please let us know.

[More about Timescale-db licensing ](https://docs.timescale.com/about/latest/timescaledb-editions/)

**Reply by tonven** [reply]
I changed postgresql.conf on selfhosted version to user community license, but it does not work. Do you have an example on how to do it?

**Reply by paulwer** [reply]
@ShanArosh @tonven @maelp have anyone found a solution for that?

**Reply by ShanArosh** [reply]
use Postgres `Replication` to copy data from supabase to timescaledb instance. you'll need to either self host or use timescaledb cloud.

**Comment by maelp** [comment]
I'm interested too in a tutorial / quick README on how to make it work with all the features?
